### PR TITLE
Disallow non-namable types for inline arrays.

### LINKED ISSUE #9004: ICE when trying to create inline arrays of some types (Solidity 0.6.8)
There are uncaught ICEs when trying to create inline arrays of some types (this is minor, since you couldn't do anything useful with such arrays anyways).

```
    [msg]; // the predefined msg struct
    [arr[0:1]]; // where arr is any kind of array
    [type(C)]; // where C is a contract
```

The two first give the following:
```
Internal compiler error during compilation:
/solidity/libsolidity/ast/Types.h(215): Throw in function virtual unsigned int solidity::frontend::Type::calldataEncodedSize(bool) const
Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

The last one is slightly different:
```
Internal compiler error during compilation:
/solidity/libsolidity/ast/Types.h(841): Throw in function virtual std::unique_ptr<solidity::frontend::ReferenceType> solidity::frontend::ArraySliceType::copyForLocation(solidity::frontend::DataLocation, bool) const
Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/9004

### CODE REVIEW

**Comment by chriseth** [comment]
Actually it turns out that the slice part is not yet fixed.

**Comment by chriseth** [comment]
Now also the slice part is "fixed".

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -1509,6 +1509,12 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 		{
 			if (!inlineArrayType)
 				m_errorReporter.fatalTypeError(6378_error, _tuple.location(), "Unable to deduce common type for array elements.");
+			else if (!inlineArrayType->namable())
+				m_errorReporter.fatalTypeError(
+					0000_error,
```
Comment (leonardoalt) [inline libsolidity/analysis/TypeChecker.cpp]: Needs to be updated

**Review by leonardoalt — CHANGES_REQUESTED** [review]
Apparently `nameable` seems to be "more right" than `namable`

**Comment by leonardoalt** [comment]
One of the tests is leading to `UnimplementedFeatureError: Conversion from calldata slices to memory not yet implemented.`

**Review by leonardoalt — DISMISSED** [review]

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**